apis/python/src/tiledbsoma/_collection.py

```diff
@@ -429,15 +429,20 @@ def __getitem__(self, key: str) -> CollectionElementType:
             from . import _factory  # Delayed binding to resolve circular import.
             from ._tdb_handles import Wrapper
 
+            uri = entry.entry.uri
             wrapper: type[Wrapper[Any | Any | Any]]
-            if self.mode == "r" and clib.SOMADataFrame.exists(entry.entry.uri):
+            if self.mode == "r" and clib.SOMADataFrame.exists(uri):
```

Comment:
I probably should have noticed this earlier, but: we probably need to be passing the context we have to this function, right? It might contain user credentials or other important configuration (e.g. endpoint locations)